### PR TITLE
avatar: add support for auto update 

### PR DESCRIPTION
**Pull request by Chilledheart**
- fix a bug with copy assignment with Account

### CODE REVIEW

File: src/avatar-service.cpp [hunk 1]
```diff
@@ -229,12 +302,14 @@ QImage AvatarService::getAvatar(const QString& email)
 {
     QImage img = loadAvatarFromLocal(email);
 
-    if (img.isNull()) {
+    // update all avatars if feature autoupdate enabled or img is null
+    if (autoupdate_db_ || img.isNull()) {
```
Comment (lins05) [inline src/avatar-service.cpp]: Each time `AvatarService::getAvatar` is called (it's called frequently when painting activities tab), a new GetAvatarRequest would be sent (if there isn't one in the queue), this is not good. We should only send a `GetAvatarRequest` either:

- We have no avatar for the current account yet, or
- We have not checked avatars for a period of time, for example 5 minutes.

**Comment by lins05** [comment]
Sometimes it fails to update the avatar even if I have updated it from seahub.

**Comment by Chilledheart** [comment]
@lins05 the seahub issue is posted internally and should be fixed now(?).

File: src/avatar-service.cpp (hunk 1 above)
Comment (Chilledheart) [inline src/avatar-service.cpp]: Thanks for the advice, I will do it sometime later.

Comment (Chilledheart) [inline src/avatar-service.cpp]: cc@lins05 Set a 5-min expire time to fix this bug.

**Comment by Chilledheart** [comment]
Sec, there is still a bug for the expire time checking.

**Pull request merged**